wavs-staking-template/dev_notes.md

```diff
@@ -0,0 +1,197 @@
+
+You need to prepare an .env file in this directory for running the rest of the commands.
+
+```bash
+cp env.example .env
+# edit the file and make any changes you need there
+```
+
+### Deploy eigen middleware
+
+```bash
+docker run --rm --network host --env-file .env  -v ./.nodes:/root/.nodes wavs-middleware:local
+export SERVICE_MANAGER_ADDRESS=$(jq -r .addresses.WavsServiceManager .nodes/avs_deploy.json)
+```
+
+### Build Custom Application
+
+Assuming you are using the template app, you can do the following. More advanced apps will have custom build steps, run from the top-level directory:
+
+```bash
+cd ..
+make setup
+forge build
+make wasi-build
+```
+
+TODO: `make wasi-build` now fails for me with `(jco componentize) ComponentError: package 'wasi:io@0.2.0' not found. no known packages.`
+
+### Deploy ServiceHandler
+
+```bash
+cd src/contracts
+forge create SimpleSubmit --broadcast --rpc-url http://127.0.0.1:8545 --private-key "$PRIVATE_KEY" --constructor-args "$SERVICE_MANAGER_ADDRESS"
+
+forge create SimpleTrigger --broadcast --rpc-url http://127.0.0.1:8545 --private-key "$PRIVATE_KEY"
+```
+
+Store "deployed_to" from the first command as `SERVICE_HANDLER_ADDRESS`.
+And store "deployed_to" from the second command as `TRIGGER_ADDRESS`.
+
+### Start aggregator
+
+In another window, Go to `aggregator` dir in `wavs-foundry-template` (you may want to edit `aggregator/.wavs_env` first)
+
+```bash
+cd wavs-staking-template/aggregator
+docker compose up
+```
+
+TODO: the following is no longer valid - we need to add the servie.json later
+
+```bash
+curl -X POST http://localhost:8001/add-service \
+     -H "Content-Type: application/json" \
+     -d "{\"eth_trigger\": {\"address\": \"$SERVICE_AGGREGATOR_ADDRESS\"}}"
+```
+
+### Create Service.JSON
+
+This file contains all the information about the service, including the service manager, the handler and trigger, as well as the aggregator address. To build your own, look at [create_service.md](create_service.md) for some info, or ask the WAVS team how to use these commands to build service.json.
+
+For deploying the template service, you can just use the template file and fill in some values, like:
+
+```bash
+cat service.json.template \
+    | sed "s/SERVICE_MANAGER_ADDRESS/$SERVICE_MANAGER_ADDRESS/g" \
+    | sed "s/SERVICE_HANDLER_ADDRESS/$SERVICE_HANDLER_ADDRESS/g" \
+    | sed "s/TRIGGER_ADDRESS/$TRIGGER_ADDRESS/g"  \
+    > service.json
+```
+
+TODO: we need to update this service.json.template to include the aggregator path as well as the service manager address. We should also push the WASI component to a registry (Layer dev team can do that once) and reference that in the template. 
+
+### Add Service.json to Service Manager
+
+TODO: upload service.json (Right now, I'm just using a git commit from this template, this should be different)
```

Comment:
noted above - we do provide a http endpoint for this on WAVS itself, but it's a stopgap measure until we have IPFS support